src/Symfony/Component/Validator/Util/DateIntervalComparisonHelper.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Util;
+
+/**
+ * @internal
+ */
+final class DateIntervalComparisonHelper
+{
+    private function __construct()
+    {
+    }
+
+    public static function supports($value, $comparedValue): bool
+    {
+        return $value instanceof \DateInterval && (
+            \is_string($comparedValue) ||
+            ($comparedValue instanceof \DateInterval && $value !== $comparedValue) ||
+            0 === $comparedValue
+        );
+    }
+
+    public static function convertValue(\DateTimeImmutable $reference, \DateInterval $value): \DateTimeImmutable
+    {
+        return \DateTimeImmutable::createFromMutable(self::getMutableReference($reference))->add($value);
+    }
+
+    public static function convertComparedValue(\DateTimeImmutable $reference, $comparedValue): \DateTimeImmutable
+    {
+        if (\is_string($comparedValue)) {
+            $reference = \DateTimeImmutable::createFromMutable(self::getMutableReference($reference));
+
+            set_error_handler(function (int $errno, string $errstr): void {
+                throw new \InvalidArgumentException($errstr);
+            });
+
+            try {
+                return $reference->modify($comparedValue);
+            } finally {
+                restore_error_handler();
+            }
+        } elseif ($comparedValue instanceof \DateInterval) {
+            return \DateTimeImmutable::createFromMutable(self::getMutableReference($reference)->add($comparedValue));
+        } elseif (0 === $comparedValue) {
```

Comment:
we could make these two "elseif" early exist statements (and the throw below too)